code/modules/atmospherics/gasmixtures/reactions.dm

```diff
@@ -433,4 +433,29 @@
 		var/new_heat_capacity = air.heat_capacity()
 		if(new_heat_capacity > MINIMUM_HEAT_CAPACITY)
 			air.temperature = max(((air.temperature*old_heat_capacity - energy_taken)/new_heat_capacity),TCMB)
-			
\ No newline at end of file
+	
+
+/datum/gas_reaction/miaster	//dry heat sterilization: clears out pathogens in the air
+	priority = -10 //after all the heating from fires etc. is done
+	name = "Dry Heat Sterilization"
+	id = "sterilization"
+
+/datum/gas_reaction/miaster/init_reqs()
+	min_requirements = list(
+		"TEMP" = FIRE_MINIMUM_TEMPERATURE_TO_EXIST+70,
+		/datum/gas/miasma = MINIMUM_MOLE_COUNT
+	)
+
+/datum/gas_reaction/miaster/react(datum/gas_mixture/air, datum/holder)
+	var/list/cached_gases = air.gases
+	// As the name says it, it needs to be dry
+	if(cached_gases[/datum/gas/water_vapor]/air.total_moles() > 0.1)
```

Comment:
You're assuming gas contains this vapor without checking first.

Reply:
Done.